### ISSUE TITLE
Automatically fetch on-chain IDL

### ISSUE THREAD

**Issue by acheroncrypto** [solana-playground/solana-playground#93]
Playground already generates UI for interacting with programs that have Anchor IDL(either built inside playground or uploaded the file manually) but it doesn't fetch on-chain IDLs.

Playground should check the program id to see if there is an Anchor IDL on-chain. This not only would allow anyone to easily interact with all programs with on-chain IDL without manually uploading it to playground, but since playground also works with localhost, it would also make the local development experience smoother.

**Comment by dhruvsol** [comment]
@acheroncrypto  the fetchIDL function in anchor can be used?

**Comment by acheroncrypto** [comment]
> @acheroncrypto the fetchIDL function in anchor can be used?

Yes, that's the best way. Ideally we would only fetch the IDL once user has put a custom program id from `Program Credentials`. And we could also add a **Fetch IDL** button under the `IDL` setting for manual fetching.
